File: src/com/demonwav/mcdev/platform/MinecraftModuleType.java
package com.demonwav.mcdev.platform;

import com.demonwav.mcdev.creator.MinecraftModuleBuilder;
import com.demonwav.mcdev.asset.PlatformAssets;

import com.intellij.openapi.module.JavaModuleType;
import com.intellij.openapi.module.ModuleTypeManager;
import org.jetbrains.annotations.NotNull;

import javax.swing.Icon;

public class MinecraftModuleType extends JavaModuleType {

    private static final String ID = "MINECRAFT_MODULE_TYPE";

    public MinecraftModuleType() {
        super(ID);
    }

    public MinecraftModuleType(String ID) {
        super(ID);
    }

    public static MinecraftModuleType getInstance() {
        return (MinecraftModuleType) ModuleTypeManager.getInstance().findByID(ID);
    }

    @NotNull
    @Override
    public MinecraftModuleBuilder createModuleBuilder() {
        return new MinecraftModuleBuilder();
    }

    @Override
    public Icon getBigIcon() {
        return PlatformAssets.MINECRAFT_ICON_2X;
    }

    @Override
    public Icon getIcon() {
        return PlatformAssets.MINECRAFT_ICON;
    }

    @Override
    public Icon getNodeIcon(@Deprecated boolean isOpened) {
        return PlatformAssets.MINECRAFT_ICON;
    }
}

File: src/main/kotlin/com/demonwav/mcdev/platform/MinecraftModuleType.kt
/*
 * Minecraft Dev for IntelliJ
 *
 * https://minecraftdev.org
 *
 * Copyright (c) 2018 minecraft-dev
 *
 * MIT License
 */

package com.demonwav.mcdev.platform

import com.demonwav.mcdev.asset.PlatformAssets
import com.demonwav.mcdev.creator.MinecraftModuleBuilder
import com.intellij.openapi.module.JavaModuleType
import com.intellij.openapi.module.ModuleTypeManager

class MinecraftModuleType : JavaModuleType() {

    override fun createModuleBuilder() = MinecraftModuleBuilder()
    override fun getIcon() = PlatformAssets.MINECRAFT_ICON
    override fun getNodeIcon(isOpened: Boolean) = PlatformAssets.MINECRAFT_ICON
    override fun getName() = NAME
    override fun getDescription() = "Minecraft modules are used for developing plugins or mods for <b>Minecraft</b> (Java Edition, also known as the PC Edition)."

    companion object {
        private const val ID = "MINECRAFT_MODULE_TYPE"
        const val NAME = "Minecraft"

        val instance: MinecraftModuleType
            get() = ModuleTypeManager.getInstance().findByID(ID) as MinecraftModuleType
    }
}
